### PR TITLE
Add error metrics for TidbCluster

### PR DESCRIPTION
**Pull request by coderplay**
### What problem does this PR solve?
This PR add fine-grained error metrics for TiDBCluster CR.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/metrics/tidbcluster.go [hunk 1]
```diff
@@ -25,4 +25,12 @@ var (
 			Name:      "spec_replicas",
 			Help:      "Desired replicas of each component in TidbCluster",
 		}, []string{LabelNamespace, LabelName, LabelComponent})
+
+	ClusterUpdateErrors = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb_operator",
+			Subsystem: "cluster",
+			Name:      "update_errors",
+			Help:      "Desired replicas of each component in TidbCluster",
```
Comment (csuzhangxc) [inline pkg/metrics/tidbcluster.go:34]: `Help` is not correct here.

Comment (coderplay) [reply pkg/metrics/tidbcluster.go]: Nice catch! Updated.

**Comment by coderplay** [comment]
example output

```
# HELP tidb_operator_cluster_update_errors Number of errors generated in each stage when updating TiDB Clusters
# TYPE tidb_operator_cluster_update_errors counter
tidb_operator_cluster_update_errors{component="pd",name="basic",namespace="default"} 1
tidb_operator_cluster_update_errors{component="tidb",name="basic",namespace="default"} 6
tidb_operator_cluster_update_errors{component="tikv",name="basic",namespace="default"} 7
```

**Comment by csuzhangxc** [comment]
@hanlins PTAL

**Review by hanlins — APPROVED** [review]
LGTM

**Comment by coderplay** [comment]
@hanlins  seems the `jenkins-ci / pull-e2e-kind-across-kubernetes` job failed due to unrelated reason.  What's the next step?

**Comment by csuzhangxc** [comment]
> @hanlins seems the `jenkins-ci / pull-e2e-kind-across-kubernetes` job failed due to unrelated reason. What's the next step?

the CI env is not stable.

**Comment by hanlins** [comment]
/cherry-pick release-1.4
cc @csuzhangxc

**Pull request merged**